File: app/Models/FeederKeypoint.php
<?php

namespace App\Models;

use Illuminate\Database\Eloquent\Model;

class FeederKeypoint extends Model
{
    public $timestamps = false;

    protected $fillable = [
        'keypoint_id',
        'feeder_id',
        'name',
    ];

    /**
     * Get the feeder that owns this keypoint.
     */
    public function feeder()
    {
        return $this->belongsTo(Feeder::class);
    }

        public function mapsData()
    {
        return $this->belongsTo(MapsData::class, 'keypoint_id', 'keypoint_id'); // atau sesuai dengan relasi yang benar
    }
}

File: go-backend-pln-monitoring/internal/model/feeder_keypoint.go
package model

// FeederKeypoint represents the feeder_keypoint model.
type FeederKeypoint struct {
	ID         uint   `json:"id" gorm:"primaryKey"`
	KeypointID string `json:"keypoint_id"`
	FeederID   uint   `json:"feeder_id"`
	Name       string `json:"name"`
}
